### PR TITLE
Fix custom path not working for docker

### LINKED ISSUE #2934: Base URL Setting for hosting in subfolders and/or behind reverse proxy
## Abstract

All URLs in Rotkis frontend are hardcoded(absolute paths)(maybe in the backend too idk). Therefore running Rotki in docker can't easily be hosted in a subfolder like https://aaa.bbb.ccc/rotki/, or served from a subfolder via a reverse proxy.

Rotkis paths should be converted to relative ones.

There should be a "baseurl" parameter for the docker container and/or Rotki itself.

## Motivation

Not every user wants to have the service beeing delivered in the domain root, as there could be other services which are served in the domain root.

## Specification
Running Rotki in a Docker container.
Serving pages and services in subfolder structures via nginx reverse proxy.

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #2934

The reason it didn't work is we forgot to inject the env when building the app.

### CODE REVIEW

**Comment by yabirgb** [comment]
This still needs some work. Seems that the frontent still tries to connect to the localhost in this example https://github.com/rotki/rotki/pull/4040#issuecomment-1029116627
image

**Comment by lukicenturi** [comment]
> This still needs some work. Seems that the frontent still tries to connect to the localhost in this example [#4040 (comment)](https://github.com/rotki/rotki/pull/4040#issuecomment-1029116627) image

Can you try again?

**Comment by kelsos** [comment]
So in the current form, the frontend gets served but not all paths work correctly.

For example, the backend still goes to `/api` instead of `/subpath/api`. There is also at least one frontend asset that is giving 404 due to being served from the root instead.

I have a few ideas on what to fix so I am gonna take it over @lukicenturi, thanks figuring out the problem. :)

**Comment by kelsos** [comment]
I am adding it here to have as a future reference.

To test the subpath one can build the image locally using the following: 
```bash
./packaging/docker-image.sh
```

This will build a `rotki:latest` image. Then you need to use the following `docker-compose.yml`, do a `docker compose up -d` to start the services. After everything is running rotki will be available under `http://localhost/rotki`.

```yaml
version: "3.9"
services:
  proxy:
    image: "traefik:v2.9"
    ports:
      - "80:80"
    command:
      - "--providers.docker.exposedbydefault=false"
      - "--entrypoints.web.address=:80"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"
  rotki:
    image: "rotki:latest"
    volumes:
      - logs:/logs
      - data:/data
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.rotki.rule=PathPrefix(`/rotki{regex:$$|/.*}`)"
      - "traefik.http.routers.rotki.middlewares=rotki-stripprefix"
      - "traefik.http.middlewares.rotki-stripprefix.stripprefix.prefixes=/rotki"
      - "traefik.http.routers.rotki.entrypoints=web"

volumes:
  logs:
  data:
```

File: frontend/app/scripts/build.js [hunk 1]
```diff
@@ -13,6 +14,14 @@ const parser = new ArgumentParser({
 parser.add_argument('--mode', { help: 'mode docker', default: 'production' });
 const { mode } = parser.parse_args();
 
+const injectEnv = (envName = '.env') => {
+  const envPath = path.resolve(__dirname, '../' + envName);
+  const envExists = fs.existsSync(envPath);
+  if (envExists) {
+    require('dotenv').config({ path: envPath, override: true });
```
Comment (kelsos) [inline frontend/app/scripts/build.js:21]: So override is needed here otherwise the first set value is used.

**Review by kelsos — APPROVED** [review]
@lukicenturi I am approving but we must do 2 different tests

1. test with the supplied config that subpath works fine without any 404s or errors
2. make sure that it is not broken for root path (running a container directly from the image

If these two scenarios work fine we can merge.

**Comment by kelsos** [comment]
@lukicenturi on my side both cases seem to work fine. 

If you can also verify we can go ahead and merge it.

**Comment by lukicenturi** [comment]
@kelsos Tried it, and nothing breaks. I'm gonna merge

**Pull request merged**